src/common/middleware/call-api.js

```diff
@@ -0,0 +1,101 @@
+import { checkStatus, getError } from '../helpers/api';
+
+// Action key that carries API call info interpreted by this Redux middleware.
+export const CALL_API = 'CALL_API';
+
+// A Redux middleware that interprets actions with CALL_API info specified.
+// Performs the call and promises when such actions are dispatched.
+export default defaults => () => next => async action => {
+  // If action does not invoke CALL_API,
+  // ignore it
+  const settings = action[CALL_API];
+  if (typeof settings === 'undefined') {
+    return next(action);
+  }
+
+  // Validate action settings
+  validateDefaults(defaults);
+  validateSettings(settings);
+
+  const [ successType, failureType ] = settings.types;
+  const resource = defaults.endpoint + settings.path;
+  const optionsWithCsrfToken = withCsrfToken(defaults.csrfToken, settings.options);
+  const createAction = createActionWith.bind({}, action);
+
+  // If action has a type, dispatch the action next
+  if (action.type) {
+    next(createAction());
+  }
+
+  try {
+    const response = await fetch(resource, optionsWithCsrfToken);
+    await checkStatus(response);
+    const result = await response.json();
+
+    if (result.status !== 'success') {
+      throw getError(response, result);
+    }
+
+    return next(createAction({
+      type: successType,
+      response: result.payload
+    }));
+  }
+  catch(error) {
+    return next(createAction({
+      type: failureType,
+      error: error
```

Comment:
Similar comment to above, in Flux Standard Action error should be sent in `payload` and `error: true` added to action object.